numpy/core/tests/test_shape_base.py

```diff
@@ -123,6 +122,59 @@ def test_3D_array(self):
         assert_array_equal(res, desired)
 
 
+class TestAtleastNd(object):
+    def test_0D_arrays(self):
+        arrays = [array(1), array(2)]
+        dims = [3, 2]
```

Comment:
Should be a test somewhere that `atleast_nd(zero_d_array, 0, 0)` is legal and return an `array` and not a scalar.